libcore/File.vala

```diff
@@ -1094,36 +1095,60 @@ public class Files.File : GLib.Object {
         return null;
     }
 
-    private void update_size () {
+    public void update_size () {
         if (is_folder () || is_root_network_folder ()) {
-            format_size = item_count ();
+            format_size = format_item_count ();
         } else if (info.has_attribute (GLib.FileAttribute.STANDARD_SIZE)) {
             format_size = GLib.format_size (size);
         } else {
             format_size = _("Inaccessible");
         }
     }
 
-    private string item_count () {
-        if (is_mounted && location.is_native ()) {
-            try {
-                var f_enum = location.enumerate_children ("", FileQueryInfoFlags.NONE, null);
-                var count = 0;
-                while (f_enum.next_file () != null) {
-                    count++;
-                }
+    private string format_item_count () {
+        ensure_item_count (true); // Re-count file items
+
+        if (count < 0) {
+            return (_("—"));
+        } else if (count == 0) {
+            return _("Empty");
+        } else {
+            return ngettext ("%'d item", "%'d items", count).printf (count);
```

Comment:
Does this text need to be translated? I am not too familiar with how ngettext works with it.

Reply:
`ngettext` does the translation.